src/ReflectionClassConstant.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2015, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+declare(strict_types=1);
+
+namespace Go\ParserReflection;
+
+use Go\ParserReflection\ValueResolver\NodeExpressionResolver;
+use PhpParser\Node\Const_;
+use PhpParser\Node\Stmt\ClassConst;
+use PhpParser\Node\Stmt\ClassLike;
+use \ReflectionClassConstant as BaseReflectionClassConstant;
+
+class ReflectionClassConstant extends BaseReflectionClassConstant
+{
+    /**
+     * Concrete class constant node
+     *
+     * @var ClassConst
+     */
+    private $classConstantNode;
+
+    /**
+     * @var Const_
+     */
+    private $constNode;
+
+    /**
+     * Name of the class
+     *
+     * @var string
+     */
+    private $className;
+
+    /**
+     * Parses class constants from the concrete class node
+     *
+     * @param ClassLike $classLikeNode Class-like node
+     * @param string $reflectionClassName FQN of the class
+     *
+     * @return array|ReflectionClassConstant[]
+     */
+    public static function collectFromClassNode(ClassLike $classLikeNode, $reflectionClassName)
```

Comment:
Please, use typehints for arguments and return values everywhere to keep code failproof

Reply:
Done!